Source/JavaScriptCore/yarr/YarrInterpreter.cpp

```diff
@@ -408,47 +490,76 @@ class Interpreter {
         return false;
     }
 
-    bool checkCharacter(int testChar, unsigned negativeInputOffset)
+    bool checkCharacter(ByteTerm& term, unsigned negativeInputOffset)
     {
-        return testChar == input.readChecked(negativeInputOffset);
+        ASSERT(term.isCharacterType());
+        if (term.matchDirection() == Forward)
+            return term.atom.patternCharacter == input.readChecked(negativeInputOffset);
+
+        if (!input.isValidOffset(negativeInputOffset))
+            return false;
+
+        return term.atom.patternCharacter == input.tryReadBackward(negativeInputOffset);
```

Comment:
Why don't we have `readBackwardChecked` and use it instead?
I see that readSurrogatePairChecked and readBackwardSurrogatePairChecked are paired. But it seems that readChecked's pair, `readBackwardChecked` doesn't exist.

Reply:
Given that any backward matching term could be preceded by a term with a variable count, we need to check before reading a character that we have input when we read backward.  Thus the name `tryReadBackward`.